alot/commands/__init__.py

```diff
@@ -15,25 +15,44 @@ class Command(object):
 
     """base class for commands"""
     repeatable = False
+    mode = ""
+    cmdname = ""
 
     def __init__(self):
         self.undoable = False
         self.help = self.__doc__
 
-        class_ = type(self)
-        class_name = class_.__name__
-        cmdname, mode = reverse_lookup_command(class_)
-
-        # fetch and set pre and post command hooks
-        # they are set to None if not defined in the hooks file
-        get_hook = settings.get_hook
-        self.prehook = (get_hook('pre_%s_%s' % (mode, cmdname)) or
-                        get_hook('pre_global_%s' % cmdname))
-        self.posthook = (get_hook('post_%s_%s' % (mode, cmdname)) or
-                         get_hook('post_global_%s' % cmdname))
+        # set to False if no attemtp to fetch hooks from settings
+        self._prehook = False 
+        self._posthook = False
```

Comment:
I had them `False` from the start since they might be set to `None` there is no hook for the particular command. That way, only a single call to `settings.get_hook()` has to be made, not that there would be very many though.